### PR TITLE
[WIP] Convert command actions tests to Javascript

### PR DESCRIPTION
**Pull request by clarmso**
This PR consists of the command actions tests from the driver package converted to Javascript from Coffeescript.  The conversion has been done via the following command:

https://github.com/cypress-io/cypress/blob/844606bcede477f18eb83834f51941e4f2f165df/.vscode/tasks.json#L15

I've converted the files in as few commits as possible.  (bulk-decaffeinate creates commits for stages of the conversion.) Please do not squash this commit. :)

A part of #2690

### User facing changelog

No user facing changelog specific to these tests should be needed.  

### Additional details

This PR is a part of continuous improvement in maintaining the codebase.  We hope the codebase will be in Typescript. :)  Javascript is an intermediate step.

I've reviewed the results of the bulk-decaffeinate command.  I've ensured the following:
* arrow functions are used if applicable.  The remaining use of the function() notation is necessary. (Example: use of "this" keyword in the tests)
* Remove unnecessary returns.
* Remove variable declarations that are unnecessary.

### How has the user experience changed?

No noticeable changes should be detected by the developers and the users.  The new version of the tests should work the same as the old Coffeescript ones.

### PR Tasks

- [X] Have tests been added/updated?
- [X] Ensure only decaffeinate + 1 cleanup commits are issued

### CODE REVIEW

**Comment by clarmso** [comment]
The test failures from server-e2e-tests-electron-3 does not seem to be related to this change. Please let me know if it is.

https://app.circleci.com/jobs/github/cypress-io/cypress/272830

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 1]
```diff
@@ -0,0 +1,1233 @@
...
+const {
+  Promise,
+} = Cypress
+
+describe('src/cy/commands/actions/check', () => {
+  before(() => {
+    cy
+    .visit('/fixtures/dom.html')
+    .then((win) => {
+      this.body = win.document.body.outerHTML
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 2]
```diff
@@ -0,0 +1,1233 @@
...
+    })
+  })
+
+  beforeEach(() => {
+    const doc = cy.state('document')
+
+    cy.$$(doc.body).empty().html(this.body)
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 3]
```diff
@@ -0,0 +1,1233 @@
+const $ = Cypress.$.bind(Cypress)
+const {
+  _,
+} = Cypress
+const {
+  Promise,
+} = Cypress
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: These are pulling from the same source, can join them.

```js
const { _, Promise } = Cypress
```

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 4]
```diff
@@ -0,0 +1,1233 @@
...
+      cy.get(':checkbox:first').check().then(() => {
+        expect(Promise.delay).to.be.calledWith(50, 'click')
+      })
+    })
+
+    describe('assertion verification', () => {
+      beforeEach(() => {
+        cy.on('log:added', (attrs, log) => {
+          if (log.get('name') === 'assert') {
+            this.lastLog = log
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 5]
```diff
@@ -0,0 +1,1233 @@
...
+          }
+        })
+      })
+
+      it('eventually passes the assertion', () => {
+        cy.$$(':checkbox:first').click(function () {
+          _.delay(() => {
+            cy.$$(this).addClass('checked')
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow function, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 6]
```diff
@@ -0,0 +1,1233 @@
...
+          }
+          , 100)
+        })
+
+        cy.get(':checkbox:first').check().should('have.class', 'checked').then(() => {
+          const {
+            lastLog,
+          } = this
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 7]
```diff
@@ -0,0 +1,1233 @@
...
+          expect(lastLog.get('state')).to.eq('passed')
+
+          expect(lastLog.get('ended')).to.be.true
+        })
+      })
+
+      it('eventually passes the assertion on multiple :checkboxs', () => {
+        cy.$$(':checkbox').click(function () {
+          _.delay(() => {
+            cy.$$(this).addClass('checked')
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 8]
```diff
@@ -0,0 +1,1233 @@
...
+    describe('errors', () => {
+      beforeEach(() => {
+        Cypress.config('defaultCommandTimeout', 100)
+
+        this.logs = []
+
+        cy.on('log:added', (attrs, log) => {
+          this.lastLog = log
+
+          this.logs.push(log)
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 9]
```diff
@@ -0,0 +1,1233 @@
...
+
+      it('throws when any member of the subject isnt visible', (done) => {
+        const chk = cy.$$(':checkbox').first().hide()
+
+        cy.on('fail', (err) => {
+          const {
+            lastLog,
+          } = this
+
+          expect(this.logs.length).to.eq(chk.length + 1)
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js [hunk 10]
```diff
@@ -0,0 +1,1233 @@
...
+
+        cy.get(':checkbox:first').check()
+      })
+
+      it('throws when subject is disabled', (done) => {
+        cy.$$(':checkbox:first').prop('disabled', true)
+
+        cy.on('fail', (err) => {
+          // get + type logs
+          expect(this.logs.length).eq(2)
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/actions/check_spec.js]: `this` with arrow functions, is this maintaining the correct `this`?

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 3 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Agree. Will fix.

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 1 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: I confirm that `this.body` is visible from the test. I've tested with inserting a `cy.log(this.body)` inside a test and the html is in the log.

Are you referring to this feature of mocha.js?
https://mochajs.org/#arrow-functions

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 4 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Yes, ditto.  I've tested with a `cy.log(this.lastLog)` in an existing test locally.

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 5 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Yes, `this` is the `<input>` element. I've inserted a `console.log(this)` locally.

`<input type="checkbox" name="colors" value="blue" class="checked">`

The `function()` cannot be replaced by an arrow function.

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 6 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Yes, `lastLog` has the value of `this.lastLog` and is not empty.

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 7 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Yes, ditto the previous test.

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 8 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Yes, the `cy.on('fail', (err) => ...` in the subsequent tests receives this.logs

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 9 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Yes.

File: packages/driver/test/cypress/integration/commands/actions/check_spec.js (hunk 10 above)
Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/actions/check_spec.js]: Yes

**Comment by clarmso** [comment]
I'd be happy to change the arrow functions back to `function()` if it's easier for everyone. Please let me know.

**Comment by jennifer-shehane** [comment]
Sorry, this got out of date so quickly. You're welcome to open a new PR.

**Pull request closed without merging**